### PR TITLE
[Image Copy] Fix the hyper_v_generation of image copy

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Fix the issue that the hyper_v_generation is always V1 when copying the image.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/image-copy/azext_imagecopy/custom.py [hunk 1]
```diff
@@ -94,13 +94,15 @@ def imagecopy(cmd, source_resource_group_name, source_object_name, target_locati
                                        '--location', snapshot_location,
                                        '--resource-group', source_resource_group_name,
                                        '--source', source_os_disk_id,
-                                       '--source-storage-account-id', source_storage_account_id])
+                                       '--source-storage-account-id', source_storage_account_id,
+                                       '--hyper-v-generation', json_cmd_output['hyperVGeneration']])
```
Comment (wangzelin007) [inline src/image-copy/azext_imagecopy/custom.py:98]: Can we add hyper_v_generation = json_cmd_output['hyperVGeneration'] after line 87
```suggestion
                                       '--hyper-v-generation', hyper_v_generation])
```

File: src/image-copy/azext_imagecopy/custom.py [hunk 2]
```diff
@@ -94,13 +94,15 @@ def imagecopy(cmd, source_resource_group_name, source_object_name, target_locati
...
     else:
         cli_cmd = prepare_cli_command(['snapshot', 'create',
                                        '--name', source_os_disk_snapshot_name,
                                        '--location', snapshot_location,
                                        '--resource-group', source_resource_group_name,
-                                       '--source', source_os_disk_id])
+                                       '--source', source_os_disk_id,
+                                       '--hyper-v-generation', json_cmd_output['hyperVGeneration']])
```
Comment (wangzelin007) [inline src/image-copy/azext_imagecopy/custom.py:105]: same as above

File: src/image-copy/azext_imagecopy/custom.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/image-copy/azext_imagecopy/custom.py]: OK, updated

**Comment by yonzhan** [comment]
Image Copy

**Review by wangzelin007 — APPROVED** [review]

**Pull request merged**